spring-pulsar/src/main/java/io/streamnative/example/datagen/SignupGenerator.java

```diff
@@ -0,0 +1,45 @@
+/**
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * <p>http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * <p>Unless required by applicable law or agreed to in writing, software distributed under the
+ * License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.streamnative.example.datagen;
+
+import com.devskiller.jfairy.Fairy;
+import com.devskiller.jfairy.producer.person.Person;
+import io.streamnative.example.model.Signup;
+import io.streamnative.example.model.SignupTier;
+import java.util.Random;
+import org.springframework.stereotype.Component;
+
+@Component
```

Comment:
To avoid bean and components in a simple app like this I would instead remove the `@Component` and just wire this in as a `@Bean` in the signup config class.